lib/Feature/FeatureModelParser.cpp

```diff
@@ -123,6 +123,32 @@ FeatureModelXmlParser::parseConfigurationOption(xmlNode *Node,
   return Ok();
 }
 
+long FeatureModelXmlParser::parseNumber(llvm::StringRef Str) {
+  if (Str.contains_lower('e')) {
+    double Number;
+    std::stringstream Stream(Str.str());
+    Stream >> Number;
+    if (Stream.fail()) {
+      llvm::errs() << "Failed to parse number '" << Str << "'\n";
+      return 0;
+    }
+    return (long)round(Number);
+  }
+
+  try {
+    return std::stol(Str.str());
+  } catch (std::invalid_argument &_) {
+    llvm::errs() << "Failed to parse number '" << Str << "'\n";
+    return 0;
+  } catch (std::out_of_range &_) {
+    llvm::errs() << "Number out of range.\n";
+    if (Str.startswith("-")) {
+      return std::numeric_limits<long>::min();
+    }
+    return std::numeric_limits<long>::max();
+  }
+}
```

Comment:
This seems to be a duplicate of ConstrantParser `parseUnsigned`, we should refactor this into a single impl